build.gradle.kts

```diff
@@ -219,110 +199,119 @@ tasks.named("wrapper") {
+tasks.register<CopyButNeverOverwrite>("jmxPassword") {
     description = "Create config/jmxremote.password from a template."
 
-    setFileMode(0600)  // passwords must be accessible only by owner
+    fileMode = 600  // passwords must be accessible only by owner
 
     // there is a template file in $JAVA_HOME/conf/management
     from(java.nio.file.Path.of(System.getProperty("java.home"), "conf", "management"))
     include("jmxremote.password.template")
-    rename("(.*).template", '$1')
+    rename("(.*).template", "$1")
     into("config")
 
     doLast {
-        logger.warn("${it.outputs.files.singleFile}/jmxremote.password:100: Edit this to set your password.")
+        logger.warn("${this.outputs.files.singleFile}/jmxremote.password:100: Edit this to set your password.")
     }
 }
 
 // Make sure the IDE prep includes extraction of natives
-ideaModule.dependsOn extractNatives
-ideaModule.dependsOn copyInMissingTemplates
+tasks.named("ideaModule") {
+    dependsOn("extractNatives", "copyInMissingTemplates")
+}
 
 // For IntelliJ add a bunch of excluded directories
 idea {
-
-    // Exclude Eclipse dirs
-    // TODO: Update this as Eclipse bin dirs now generate in several deeper spots rather than at top-level
-    module.excludeDirs += file("bin")
-    module.excludeDirs += file(".settings")
-    // TODO: Add a single file exclude for facades/PC/Terasology.launch ?
-
-    // Exclude special dirs
-    module.excludeDirs += file("natives")
-    module.excludeDirs += file("protobuf")
-
-    // Exclude output dirs
-    module.excludeDirs += file("configs")
-    module.excludeDirs += file("logs")
-    module.excludeDirs += file("saves")
-    module.excludeDirs += file("screenshots")
-    module.excludeDirs += file("terasology-server")
-    module.excludeDirs += file("terasology-2ndclient")
-
-    module.downloadSources = true
+    module {
+        excludeDirs = setOf(
+            // Exclude Eclipse dirs
+            // TODO: Update this as Eclipse bin dirs now generate in several deeper spots rather than at top-level
+            file("bin"),
+            file(".settings"),
+            // TODO: Add a single file exclude for facades/PC/Terasology.launch ?
+
+            // Exclude special dirs
+            file("natives"),
+            file("protobuf"),
+
+            // Exclude output dirs
+            file("configs"),
+            file("logs"),
+            file("saves"),
+            file("screenshots"),
+            file("terasology-server"),
+            file("terasology-2ndclient")
+        )
+        isDownloadSources = true
+    }
 
     project.settings.delegateActions {
         delegateBuildRunToGradle = false
         testRunner = ActionDelegationConfig.TestRunner.CHOOSE_PER_TEST
     }
 }
 
-cleanIdea.doLast {
-    new File("Terasology.iws").delete()
+tasks.register("cleanIdeaIws") {
+    doLast {
+        File("Terasology.iws").delete()
+    }
+}
+
+tasks.named("cleanIdea") {
+    dependsOn("cleanIdeaIws")
```

Comment:
it was so that one calls "cleanIdea" task and it cleaned Terasology.iws. this is in kotlin like this as well. i made a task as i did not know another way to add deleting this file to existing task "cleanIdea".

apart from idea tasks not working for me in groovy, and not in ktolin, but this is only kotlin to groovy. what is there before is there after.